moved subman_ok.feature to the features/rhelah directory

ljozsa opened the pull request.

Commit:
moved subman_ok.feature to the features/rhelah directory

Commit:
Bad username test for cloud-init subman plugin

Commit:
Bad password test for cloud-init subman plugin

Commit:
steps/rhelah.py: removed whitespace from matched string

Commit:
subman: test whether attaching non-existent pool fails

Commit:
subman: test whether attaching scalar pool fails
steps/rhelah.py: fixed typo in non-existent pool-id test

aweiteka commented:
@ljozsa could you move some of the `subscription-manager` tests to the `[redhat.py](https://github.com/aweiteka/UATFramework/blob/master/steps/redhat.py)` steps file?

aweiteka commented on steps/rhelah.py:
```diff
@@ -317,3 +317,33 @@ def step_impl(context):
                     module_args='grep cc_rh_subscription.py /var/log/cloud-init.log | grep "Enabled the following repos" | cut -d ":" -f5 | sed "s/^ //"')[0]['stdout']
 	assert repoids_enabled == 'rhel-7-server-optional-beta-rpms, rhel-7-server-beta-debug-rpms', "Configured repoids weren't enabled"
+
+@then(u'check if it failed')
```
These are global names. Could you make this more specific to the target action?

ljozsa commented:
@aweiteka I'll move them in the next commit. I'm preparing several other commits so please leave the pull request open. I'll let you know as soon as I finish this batch of commits.

Commit:
moved subman steps to the steps/redhat.py

Commit:
subman: test of existent/non-existent pools mix
 steps/redhat.py, rhelah/subman_ok.feature: slightly reformulated text of the test in order to play nicely with the existent/non-existent pools mix test

Commit:
subman: test adding non-existent repoid

Commit:
subman: test enabling/disabling already enabled/disabled repoids

Commit:
subman: added non-existent keys test

ljozsa commented:
@aweiteka I'm done with commits. You can go ahead with reviewing/merging the pull request.

miabbott commented on steps/redhat.py:
```diff
@@ -72,3 +72,95 @@ def step_impl(context, host):
+                    module_args='grep cc_rh_subscription.py /var/log/cloud-init.log | grep "Enabled the following repos" | cut -d ":" -f5 | sed "s/^ //"')[0]['stdout']
+	assert repoids_enabled == 'rhel-7-server-optional-beta-rpms, rhel-7-server-beta-debug-rpms', "Configured repoids weren't enabled"
+
+@then(u'check if it failed')
```
This is too generic.  Check if what failed?

miabbott commented on steps/redhat.py:
```diff
@@ -72,3 +72,95 @@ def step_impl(context, host):
+
+	assert cloudinit_completed[0].has_key('failed') == False, "The cloud-init service did not complete"
+
+@then(u'check if it completed successfully')
```
This is too generic.  Check if what completed successfully?

miabbott commented on steps/redhat.py:
```diff
@@ -72,3 +72,95 @@ def step_impl(context, host):
+                    module_args='grep cc_rh_subscription.py /var/log/cloud-init.log | grep Repo | grep exist | cut -d ":" -f4 | sed -e "s/^ //"')[0]['stdout']
+	assert register_result == 'Repo rhel-7-server-beta-debug-rpm does not appear to exist', "Error message not found"
+
+@then(u'check the already enabled message')
```
I would make this more specific...along the lines of "check for the repo already enabled message"

miabbott commented on steps/redhat.py:
```diff
@@ -72,3 +72,95 @@ def step_impl(context, host):
+                    module_args='grep cc_rh_subscription.py /var/log/cloud-init.log | grep Repo | grep already | cut -d ":" -f4 | sed -e "s/^ //"')[0]['stdout']
+	assert register_result == 'Repo rhel-rs-for-rhel-7-server-eus-rpms is already enabled', "Informational error message not found"
+
+@then(u'check the already disabled message')
```
I would make this more specific...along the lines of "check for the repo already disabled message"

miabbott commented:
How is the bad input (bad password, bad pool-id) being provided for these tests?  During the provisioning stage?

miabbott commented on steps/redhat.py:
```diff
@@ -72,3 +72,95 @@ def step_impl(context, host):
+                    module_args='grep cc_rh_subscription.py /var/log/cloud-init.log | grep Regist | cut -d ":" -f4 | sed -e "s/^ //" -e "s/ [-a-f0-9]\+//" -e "s/ $//"')[0]['stdout']
+	assert register_result == 'Registered successfully with ID', "subscription-manager did not register successfully"
+
+@then(u'check if it successfully attached existing pools')
```
Too generic.  Check if what successfully attached existing pools?

Commit:
Made too generic tests more specific

ljozsa commented:
@miabbott I have replaced generic sentences with more specific ones. Please check it out.

miabbott commented on features/rhelah/subman_ok.feature:
```diff
@@ -16,7 +16,7 @@ Background: Atomic hosts are discovered
   Scenario: 2. subscription-manager plugin attaches existing pools
        Given cloud-init on "all" host is running
-       Then check if it successfully attached defined pools
+       Then check if it successfully attached existing pools
```
This is still too generic, in my opinion.  Please update this line and the associated step here:

https://github.com/aweiteka/UATFramework/pull/43/files#diff-6b8f0fd24b0008acc91407983ef6f088R102

miabbott commented:
@ljozsa Good improvements.  Just one last piece of generic code that I commented on.

Commit:
Poolid check rewritten to be more specific

ljozsa commented:
@miabbott OK, rewritten. Check it out.

miabbott commented:
@ljozsa Sorry, more requests from me.  :disappointed:

Commit:
Poolid check rephrased once more

ljozsa commented:
@miabbott No problem Micah :-) I've rephrased it once again.

miabbott commented on steps/redhat.py:
```diff
@@ -72,3 +72,95 @@ def step_impl(context, host):
+                    module_args='grep cc_rh_subscription.py /var/log/cloud-init.log | grep Repo | grep exist | cut -d ":" -f4 | sed -e "s/^ //"')[0]['stdout']
+	assert register_result == 'Repo rhel-7-server-beta-debug-rpm does not appear to exist', "Error message not found"
+
+@then(u'check the Repo rhel-rs-for-rhel-7-server-eus-rpms is already enabled message appearance')
```
In order to make this step more modular, could you make the repo name a variable?  (This would also apply to the step at lines 156-160.  And the feature file would need to be updated.)

For example:

``` python
@then(u'check the Repo "{reponame}" is already enabled message appearance')
def step_impl(context):
    register_result =  context.remote_cmd(cmd='shell',
                   module_args='grep cc_rh_subscription.py /var/log/cloud-init.log | grep Repo | grep already | cut -d ":" -f4 | sed -e "s/^ //"')[0]['stdout']
        repo_message = 'Repo ' + reponame + ' is already enabled'
    assert register_result == repo_message, "Informational error message not found"
```

miabbott commented:
@ljozsa Another suggestion.  :smiley:

Commit:
Re-enable/re-disable test: made repo name variable

ljozsa commented:
@miabbott Changes commited.

miabbott commented:
Thanks @ljozsa!  I'm merging this.
